### ISSUE TITLE
`interopDefault` option doesn't work the same past 2.2.0

### ISSUE THREAD

**Issue by lachieh** [unjs/jiti#341]
### Environment

See reproduction.

### Reproduction

https://stackblitz.com/edit/jiti-interopdefault-testing?file=src%2Fmain.ts

### Describe the bug

In upgrading to v2 for [docusaurus](https://github.com/facebook/docusaurus/pull/10716), I'm discovering some odd behavior.

It seems that the Proxy implementation for interopDefault may not be entirely working as expected. In the repro link, you can see that the properties are accessible, but only when directly accessing the properties (since that's how Proxies work). The issue is that this is seemingly causing issues with testing.

**Comment by pi0** [comment]
Thanks for explaining the issue. I understand that the new behavior affects testing when modules are strictly being compared as objects.

Unfortunately, it is not easy to avoid proxy, as it covers many edge cases in mixed CJS/ESM modules.

I'm wondering is it affecting any real usecase in docusaurus too with wrong behavior of proxy?

**Comment by lachieh** [comment]
Yeah, I'm investigating that now. It seems like it does since the E2E tests were failing, specifically when importing the main config file in esm format. I actually just ended up wrapping the imported proxy in a `mlly.interopDefault` call and it solved 99% of the test failures but that feels a little strange.

The last remaining test failure seems to be something to do with a nested dependency not returning the proxy object. The usage of jiti is [here](https://github.com/facebook/docusaurus/blob/bd88110d417ef357a592ebc218680302abb4798d/packages/docusaurus-utils/src/moduleUtils.ts#L14-L45). If you can see anything that might cause the dependencies to skip the jiti import, I'm all ears.

**Comment by pi0** [comment]
Are both modules and configs having a default export? In this case, you might want to use `jiti.import(id, { default: true })` (we use it internally for c12, config loader of Nuxt and Nitro)

**Comment by lachieh** [comment]
No, plugins use named and default exports, and configs (typically) have a default export.

For more detail, the larger test logic looks like this:

1. Test runs, creating the `entryFile`, and some dependency files with the `fileHelper` function:
   https://github.com/facebook/docusaurus/blob/6aa92ae28dda326bee6ec0de1ed493e068b022ca/packages/docusaurus-utils/src/__tests__/moduleUtils.test.ts#L118-L131
1. `entryFile.load()` calls the `loadFreshModule` function which returns the `jiti.import()`'ed module.
  https://github.com/facebook/docusaurus/blob/6aa92ae28dda326bee6ec0de1ed493e068b022ca/packages/docusaurus-utils/src/__tests__/moduleUtils.test.ts#L25-L34
1. the resulting import can see the named exports on the `entryFile`, but not on the imported dependencies.
   https://github.com/facebook/docusaurus/blob/bd88110d417ef357a592ebc218680302abb4798d/packages/docusaurus-utils/src/__tests__/moduleUtils.test.ts#L134-L143

**Comment by pi0** [comment]
Surely feel free to make a PR to make `interopDefault` programmatically configurable. 

Consider that `interopDefault` is used in all intermediate modules. `mlly.interopDefault` has known issues and alone is not a safe replacement for jiti v2.

---

I have another proposal. We could add a symbol to the jiti's interop proxy that allows accessing the raw module value to manually mix. Currently something like this is also possible and I think should have similar effect (except that has one additional `default`)

```js
const { default: _default, ...namedExports } = entryWithDeps;
const mixedExports = { ...entryWithDeps }
```

**Comment by lachieh** [comment]
I like that second option much better. Would you want another option in `jiti.import()` like `default`? Perhaps `jiti.import('./file', { raw: true })`?

**Comment by pi0** [comment]
We can do that too! Or what about `default: false` that _skips_ interop default for final exports result? (currently it is `true` or nothing)

**Comment by pi0** [comment]
`Symbol.for('jiti.raw')` can be directly used in docusaurus to access raw module. It is not unsafe just not interop version of module namespace. I agree might be better to keep it mainly as a workaround (and later if common enough we can think of a flag)

**Comment by lachieh** [comment]
I'm going to close this one out. Didn't see much from other users and I'm not having the issue anymore. If anyone else comes across this, probably best to open a new issue.
